diffblue/cover-client-js
# Use official bluebird types, and make them a production dependency [TG-10031]

### Context/purpose

Users consuming this library in TS will need access to the bluebird types, since they are referenced in out build `d.ts` files  `writeTests.d.ts`

### Implementation details

Make bluebird types a production dependency and get rid of our hard-coded types for bluebird (we don't use there in Platform anymore)

### QA instructions

Install cover client in a type script project (which doesn't have `@types/bluebird` otherwise installed), you should not encounter a compiler error.

### Any unrelated changes?



### PR Checklist



- [ ] I have given the PR a meaningful title which can be understood out of context (used in the changelog)
- [ ] I have read through my code and fixed the indentation, followed naming conventions and removed leftover debugging
- [ ] I have added the Jira ticket number(s) to the title with the format "Description of the feature [TG-123, TG-456]" and have copied the QA instructions to the Jira ticket (if one exists)
